domain/ide/workspace/workspace.go

```diff
@@ -107,6 +107,15 @@ func (w *Workspace) GetFolderContaining(path string) (folder *Folder) {
 	return nil
 }
 
+func (w *Workspace) Folders() (folder []*Folder) {
+	folders := make([]*Folder, 0)
```

Comment:
Probably not critical, but this can be optimized because we know the exact length of the array.

You can `make([]*Folder, len(w.folders))`. This is faster because append keeps resizing/recreating arrays.

Reply:
Nice catch, will update :)